jina/types/document/graph.py

```diff
@@ -195,6 +209,9 @@ def remove_edge(self, doc1: 'Document', doc2: 'Document'):
             if row.item() == offset1 and col.item() == offset2:
                 self._remove_edge_id(edge_id, f'{doc1.id}-{doc2.id}')
 
+            if self.undirected:
```

Comment:
I think this is very dangerous. U would be removing the same edge twice?

Reply:
I think there are two things we can do (both of which are reasonable):


- we only work with halve the adjacency matrix and we convert `i->j` to `j->i` when needed 

- we consider every undirected edge `a-b`  is in bijection with `a->b` and ` b->a` and therefore, removing `a-b` would mean removing `a->b` and `b->a`.

Reply:
I think that since there is no computational capacity in our side ( no operations are run in this Graph), I would maybe just store half of it, and flag it as undirected. Does this make sense?

Reply:
Solved in the last commit: yes we essentially store only one edge (ordered lexicographically).